sunpy/timeseries/sources/goes.py

```diff
@@ -218,6 +218,7 @@ def _parse_hdus(cls, hdulist):
                 raise ValueError("Date not recognized")
             xrsb = hdulist[2].data['FLUX'][0][:, 0]
             xrsa = hdulist[2].data['FLUX'][0][:, 1]
+            # TODO how to extract quality flags from HDU?
```

Comment:
Do you have a FITS file to test with? I would hope it's one of the array rows in a HDU.

Reply:
When looking at the FITS header the following information is contained,
e.g. for https://umbra.nascom.nasa.gov/goes/fits/2019/go1520191210.fits:

```
import sunpy.io
from sunpy.util.metadata import MetaDict
from collections import OrderedDict

filepath= "./go1520191210.fits"
hdulist = sunpy.io.read_file(filepath)
header = MetaDict(OrderedDict(hdulist[3].header))
print(header)
print(hdulist[3].data)
>FITS_rec([(0.55999994, [0., 0.])],
         dtype=(numpy.record, [('Time', '>f4'), ('Status', '>f4', (2,))]))
```

The following doc is provided for `Status`, however it is not clear to me how the status maps to xrsa and xrsb (e.g. in netcdf format there is a value for each channel and measurement):

Convert status words to long integer then examine bits.Status 
1 - octal mask, description (incomplete list):
- 1000, Sun eclipsed by MoonStatus 
2 - octal mask, description (incomplete list):
- 1, X-Ray detector off
- 2, X-Ray detector being calibrated
- 4, X-Ray Transient
- 10, X-Ray short channel saturation
- 20, X-Ray long channel range change
- 40, X-Ray short channel range change
- 200, X-Ray long channel saturation')

Does someone know how to extract this information?

Reply:
I think this is beyond scope of this PR - could someone open an issue so we can keep track of this though?

Reply:
I think this is beyond scope of this PR - could someone open an issue so we can keep track of this though?

Reply:
I would rather we didn't add support only for one file type but I don't want to block this PR on that.